plugins/flatpak/gs-flatpak.c

```diff
@@ -3679,72 +3680,64 @@ app_has_local_source (GsApp *app)
 }
 
 static guint64
-get_installation_dir_free_space (GsFlatpak *self)
+get_installation_dir_free_space (GsFlatpak *self, GError **error)
 {
 	g_autoptr (GFile) installation_dir = NULL;
 	g_autoptr (GFileInfo) info = NULL;
-	g_autoptr (GError) local_error = NULL;
-	guint64 free_space = 0;
 
 	installation_dir = flatpak_installation_get_path (self->installation);
 
 	info = g_file_query_filesystem_info (installation_dir,
 					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
 					     NULL,
-					     &local_error);
-	if (info == NULL)
-		g_warning ("Failed to query filesystem info: %s", local_error->message);
+					     error);
+	if (info == NULL) {
+		g_warning ("Failed to query filesystem info");
```

Comment:
Since we now have an error, instead of using just as a sentinel, we should print also when happened. So the warning should include `: %s", error->message` at the end. And it's better in this case to move the warning to the caller, so you can use a specific message ("error updating", "error installing", etc.) which also prevents us from having to deal with whether the error is passed as `NULL` at some point in the future.